build.gradle

```diff
@@ -0,0 +1,157 @@
+}
+
+compileTestJava {
+    options.compilerArgs << '-Xlint:deprecation'
+}
+
+repositories {
+    maven {
+        url "https://dl.bintray.com/hmcts/hmcts-maven"
+    }
+    jcenter()
+    mavenLocal()
+}
+
+jar {
+    manifest {
+        attributes 'Implementation-Title': project.name, 'Implementation-Version': project.version
+    }
+}
+
+task printVersion {
+    doLast {
+        print project.version
+    }
+}
+
+def springBootVersion = '1.5.7.RELEASE'
+
+def pomConfig = {
+    licenses {
+        license {
+            name "MIT License"
+            url "http://www.opensource.org/licenses/mit-license.php"
+            distribution "repo"
+        }
+    }
+
+    scm {
+        url "https://github.com/hmcts/service-authorisation-access-client"
+    }
+}
+
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+artifacts {
+    archives sourcesJar
+    archives javadocJar
+}
+
+publishing {
+    publications {
+        Main(MavenPublication) {
+            from components.java
+            artifact sourcesJar
+            artifact javadocJar
+            groupId project.group
+            artifactId 'service-authorisation-access-client'
+            version project.version
+
+            pom.withXml {
+                def root = asNode()
+                root.appendNode('description', 'Client library for communicating with idam service authorisation access')
+                root.appendNode('name', 'Service authorisation access client')
+                root.appendNode('url', 'https://github.com/hmcts/service-authorisation-access-client')
+                root.children().last() + pomConfig
+            }
+        }
+    }
+}
+
+bintray {
+    user = System.getenv('BINTRAY_USER')
+    key = System.getenv('BINTRAY_KEY')
+    publications = ['Main']
+    publish = true
+    pkg {
+        repo = 'hmcts-maven'
+        name = 'service-authorisation-access-client'
+        userOrg = 'hmcts'
+        licenses = ['MIT']
+        vcsUrl = 'https://github.com/hmcts/service-authorisation-access-client'
+        version {
+            name = project.version
+        }
+    }
+}
+
+ext {
+    springCloudVersion = 'Dalston.SR4'
+}
+
+dependencies {
+    compile('org.springframework.cloud:spring-cloud-starter-feign')
+    compile('com.netflix.feign:feign-jackson:8.18.0')
+    compile group: 'com.google.guava', name: 'guava', version: '23.0'
+    compile group:'com.warrenstrange', name: 'googleauth', version: '1.1.2'
+    testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: springBootVersion
+    testCompile group: 'org.assertj', name: 'assertj-core', version: '3.8.0'
+    testCompile group: 'com.marvinformatics.feign', name: 'feign-mock', version: '0.6'
+    testCompile group: 'io.github.openfeign', name: 'feign-gson', version: '9.4.0'
```

Comment:
jackson at runtime and gson in tests?

Reply:
removed, we add more testing later when I have enough confidence testing feign clients